scripts/auth.ps1

```diff
@@ -0,0 +1,46 @@
+param (
+    [string]$server = "https://localhost:5001"
+)
+
+# Ignoring self-signed certificates
+add-type @"
+    using System.Net;
+    using System.Security.Cryptography.X509Certificates;
+    public class TrustAllCertsPolicy : ICertificatePolicy {
+        public bool CheckValidationResult(
+            ServicePoint srvPoint, X509Certificate certificate,
+            WebRequest request, int certificateProblem) {
+            return true;
+        }
+    }
+"@
+[System.Net.ServicePointManager]::CertificatePolicy = New-Object TrustAllCertsPolicy
+
+$username = Read-Host 'Please enter your username'
+$password = Read-Host 'Please enter your password' -AsSecureString
+$body = @{
+    Username=$username
+    Password=[Runtime.InteropServices.Marshal]::PtrToStringAuto([Runtime.InteropServices.Marshal]::SecureStringToBSTR($password))
+} | ConvertTo-Json
+$headers = @{
+    "Content-Type" = "application/json"

```

Comment:
The PowerShell code convention insist on using single quotas instead of double quotas if there are no escape sequences or variables. The reason for this is that single quotas disable the expansion of variables, which will reduce the unexpected results.

Reply:
Changed, but one more check would be appreciated. + Could you please elaborate where I should and shouldn't use single, double quotas.

Reply:
Sure, the main suggestion behind not using double quotas everywhere stands for it possible unexpected behavior. In case you are using double quotas you are actually allowing your script to modify the literal inside the quotas with some logic. Lets make an example:
For instance, you have a urlDomain variable $urlDomain = 'https://mydomain' and in case you are using double quotas you can do the following
$urlDomain = 'https://mydomain
"The url domain is $urlDomain" // Output: The url domain is https://mydomain
However in case you do use single quota such logic would not be possible
$urlDomain = 'https://mydomain
'The url domain is $urlDomain' // Output: The url domain is $urlDomain
Also for instance you have some numeric operations inside your string
"The value of $(1+1) is 2" //Output: The value of 2 is 2
And with single quotas
"The value of $(1+1) is 2" //Output: The value of $(1+1) is 2
So basically, you should have a rule - if you do not want to have any logic or variable value inside your string, do not use double quotas, and use single one instead!